src/dfuzzer.c

```diff
@@ -926,120 +926,53 @@ void df_print_process_info(int pid)
 {
         char proc_path[15+DECIMAL_STR_MAX(int)]; // "/proc/(int)/[exe|cmdline]"
         char name[PATH_MAX];           // for storing process and package name
-        char buf[PATH_MAX + MAXLEN];   // buffer for rpm/dpkg request
-        FILE *fp;
-        char *c;
-        int fd, stdoutcpy, stderrcpy, ret;
-
+        _cleanup_close_ int fd = -1;
+        int ret;
 
         sprintf(proc_path, "/proc/%d/exe", pid);
-        ret = readlink(proc_path, name, sizeof(name));
-        // excludes interprets
-        if (ret != -1 && strstr(name, "python") == NULL && strstr(name, "perl") == NULL)
-                fprintf(stderr, "%s%s[PROCESS: %s%s\n",
-                        ansi_cr(), ansi_cyan(), name, ansi_normal());
-        else {
-                // if readlink failed or executable was interpret (and our target is
-                // interpreted script), try to read cmdline
-                sprintf(proc_path, "/proc/%d/cmdline", pid);
-                fd = open(proc_path, O_RDONLY);
-                if (fd == -1)
-                        return;
-
-                ret = 1;
-                c = name;
-                fprintf(stderr, "%s%s[PROCESS: ", ansi_cr(), ansi_cyan());
-                while (ret > 0) {
-                        ret = read(fd, c, 1);
-                        if (ret > 0) {
-                                if (*c == '\0')
-                                        *c = ' ';
-                                fprintf(stderr, "%c", *c);
-                        }
-                        if (ret == -1) {
-                                fprintf(stderr, "%s\n", ansi_normal());
-                                close(fd);
-                                return;
-                        }
-                        c++;
-                }
-                *c = '\0';
-                fprintf(stderr, "%s\n", ansi_normal());
-                close(fd);
+        ret = readlink(proc_path, name, PATH_MAX);
+        if (ret > 0) {
+                if (ret == PATH_MAX) {
+                        df_verbose("The process name was truncated\n");
+                        name[PATH_MAX - 1] = '\0';
+                } else
+                        name[ret] = '\0';
 
-                // excludes interprets
-                if (strstr(name, "python") != NULL || strstr(name, "perl") != NULL) {
-                        fprintf(stderr, "%s%s[PACKAGE: %s\n", ansi_cr(), ansi_cyan(), ansi_normal());
+                if (!strstr(name, "python") && !strstr(name, "perl")) {
+                        fprintf(stderr, "%s%s[PROCESS: %s%s\n",
```

Comment:
I know it came from `dfuzzer` but looking at `[PROCESS: /usr/lib/systemd/systemd-resolved` I think either `[` should be removed or `]` should be appended.

Reply:
Yeah, the same thing is spread across the whole codebase and it's bugging me as well. Not sure what was the original rationale :-)